apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1293 @@
+			dest = ws_test_wget_strcpy(dest, (char *)param->entity, param);
+			if (dest == NULL) {
+				return WGET_MSG_CONSTRUCT_ERR;
+			}
+		} else { /* chunked param->encoding */
+			post_len = strlen(param->entity);
+			if (post_len > param->buflen) {
+				dest = ws_test_wget_chunksize(dest, param->buflen, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strlencpy(dest, (char *)param->entity, param->buflen, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+			} else {
+				temp = dest;
+				dest = ws_test_wget_chunksize(dest, post_len, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, (char *)param->entity, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				g_normal_clen = dest - temp;
+				dest = ws_test_wget_strlencpy(dest, "0", 1, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				if (g_trailer) {
+					dest = ws_test_wget_strcpy(dest, g_expires, param);
+					if (dest == NULL) {
+						return WGET_MSG_CONSTRUCT_ERR;
+					}
+				}
+				dest = ws_test_wget_strcpy(dest, g_httpcrnl, param);
+				if (dest == NULL) {
+					return WGET_MSG_CONSTRUCT_ERR;
+				}
+				g_last_clen = dest - temp;
+			}
+		}
+	}
+	FUNC_EX;
+	return dest - param->buffer;
+}
+
+static int ws_test_wget_socket_connect(struct ws_test_wget_s *ws)
+{
+	FUNC_EN;
+	int sockfd, ret;
+	struct timeval tv;
+	struct sockaddr_in server;
+
+	/* Create a socket */
+	sockfd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	if (sockfd < 0) {
+		return WGET_SOCKET_CONNECT_ERR;
+	}
+
+	/* Set send and receive timeout values */
+	tv.tv_sec = WEBCLIENT_CONF_TIMEOUT_MSEC / 1000;
+	tv.tv_usec = (WEBCLIENT_CONF_TIMEOUT_MSEC % 1000) * 1000;
+	PRNT("recv timeout is set to (%d.%d)sec", tv.tv_sec, tv.tv_usec);
+	if (setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO,
+				   (struct timeval *)&tv, sizeof(struct timeval)) < 0) {
+		PRNT("ERROR: setsockopt failed");
+	}
+
+	/* Get the server address from the host name */
+	server.sin_family = AF_INET;
+	server.sin_port = htons(ws->port);
+	ret = ws_test_wget_gethostip(ws->hostname, &server.sin_addr.s_addr);
+	if (ret < 0) {
+		/* Could not resolve host (or malformed IP address) */
+		PRNT("ERROR: Failed to resolve hostname");
```

Comment:
I think it is better to print the return value where from the error, because you returned the -errno for failure case.
Then it is much better to debug.

Reply:
Changes done as suggested